proto/caraml/upi/v1/observation_log.proto

```diff
@@ -46,3 +47,105 @@ message LogObservationsRequest {
 message LogObservationsResponse {
     string observation_batch_id = 1;
 }
+
+// Data source where Observation logs should be retrieved from
+enum ObservationServiceDataSourceType {
+  OBSERVATION_SERVICE_DATA_SOURCE_TYPE_UNSPECIFIED = 0;
+  // No-Op represents no need to fetch logs from any data source, this should be selected if
+  // Observation Service should be deployed for just the eager API
+  OBSERVATION_SERVICE_DATA_SOURCE_TYPE_NOOP = 1;
```

Comment:
Would `OBSERVATION_SERVICE_DATA_SOURCE_TYPE_API` (or `_EAGER`) be a better representation then? When the source is Kafka, do we also plan to make the API endpoints available?

Reply:
>Would OBSERVATION_SERVICE_DATA_SOURCE_TYPE_API (or _EAGER) be a better representation then?

Yup `_EAGER` would be a better option here.

>When the source is Kafka, do we also plan to make the API endpoints available?

Yes, API endpoints should be available for all deployments.